### PR TITLE
Add 'Bucket.default_kms_encryption_key' property.

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: storage/google/cloud/storage/bucket.py [hunk 1]
```diff
@@ -786,6 +786,32 @@ def cors(self, entries):
         """
         self._patch_property('cors', entries)
 
+    @property
+    def default_kms_encryption_key(self):
+        """Retrieve or set default KMS encryption key for items in the bucket.
+
+        See https://cloud.google.com/storage/docs/json_api/v1/buckets
+
+        :setter: Set default KMS encryption key for items in this bucket.
```
Comment (frankyn) [inline storage/google/cloud/storage/bucket.py]: `for objects in the bucket`

File: storage/google/cloud/storage/bucket.py [hunk 2]
```diff
@@ -786,6 +786,32 @@ def cors(self, entries):
...
+
+        :rtype: str
+        :returns: Default KMS encryption key, or ``None`` if not set.
+        """
+        encryption_config = self._properties.get('encryption', {})
+        return encryption_config.get('defaultKmsKeyName')
+
+    @default_kms_encryption_key.setter
+    def default_kms_encryption_key(self, value):
+        """Set default KMS encryption key for items in the bucket.
```
Comment (frankyn) [inline storage/google/cloud/storage/bucket.py]: `for objects in the bucket`

**Comment by tseaver** [comment]
@frankyn I planned to do the system tests in an additional PR (once we ironed out the API surface).

**Comment by frankyn** [comment]
Thanks for clarifying!

File: storage/google/cloud/storage/bucket.py (hunk 1 above)
Comment (tseaver) [reply storage/google/cloud/storage/bucket.py]: df91f36

File: storage/google/cloud/storage/bucket.py (hunk 2 above)
Comment (tseaver) [reply storage/google/cloud/storage/bucket.py]: df91f36

**Comment by tseaver** [comment]
@frankyn I also realized that `default_kms_encryption_key` was a poor name for the property:  it should match the API name, which fits the semantics better (not always the case :). dad38b4 updates it to `default_kms_key_name`.

**Comment by tseaver** [comment]
@dpebot please merge when green.

**Pull request merged**